arangodb/src/main/java/com/yahoo/ycsb/db/ArangoDBClient.java

```diff
@@ -0,0 +1,488 @@
+
+      // Set the DB address
+      String ip = props.getProperty("arangodb.ip", "localhost");
+      String portStr = props.getProperty("arangodb.port", "8529");
+      int port = Integer.parseInt(portStr);
+
+      // If clear db before run
+      String dropDBBeforeRunStr = props.getProperty("arangodb.dropDBBeforeRun", "false");
+      dropDBBeforeRun = Boolean.parseBoolean(dropDBBeforeRunStr);
+      
+      // Set the sync mode
+      String waitForSyncStr = props.getProperty("arangodb.waitForSync", "false");
+      waitForSync = Boolean.parseBoolean(waitForSyncStr);
+      
+      // Set if transaction for update
+      String transactionUpdateStr = props.getProperty("arangodb.transactionUpdate", "false");
+      transactionUpdate = Boolean.parseBoolean(transactionUpdateStr);
+      
+      // Init ArangoDB connection
+      arangoConfigure = new ArangoConfigure();
+      arangoConfigure.setArangoHost(new ArangoHost(ip, port));
+      arangoConfigure.init();
+      arangoDriver = new ArangoDriver(arangoConfigure);
+
+      // Init the database
+      if (dropDBBeforeRun) {
+        // Try delete first
+        try {
+          arangoDriver.deleteDatabase(databaseName);
+        } catch (ArangoException e) {
+          if (e.getErrorNumber() != ErrorNums.ERROR_ARANGO_DATABASE_NOT_FOUND) {
+            logger.error("Failed to delete database: " + databaseName, e);
+            System.exit(-1);
+          } else {
+            logger.info("Fail to delete DB, already not exists: {}", databaseName);
+          }
+        }
+      }
+      try {
+        arangoDriver.createDatabase(databaseName);
+        logger.info("Database created: " + databaseName);
+      } catch (ArangoException e) {
+        if (e.getErrorNumber() != ErrorNums.ERROR_ARANGO_DUPLICATE_NAME) {
+          logger.error("Failed to create database: " + databaseName, e);
+          System.exit(-1);
+        } else {
+          logger.info("DB already exists: {}", databaseName);
+        }
+      }
+      // Always set the default db
+      arangoDriver.setDefaultDatabase(databaseName);
+      logger.info("ArangoDB client connection created to " + ip + ":" + port);
+      
+      // Log the configuration
+      StringBuilder configStr = new StringBuilder("Arango Configuration: ");
+      configStr.append(String.format("dropDBBeforeRun: %s; ", dropDBBeforeRun));
+      configStr.append(String.format("address: %s:%s; ", ip, port));
+      configStr.append(String.format("databaseName: %s; ", databaseName));
+      configStr.append(String.format("waitForSync: %s; ", waitForSync));
+      configStr.append(String.format("transactionUpdate: %s; ", transactionUpdate));
+      logger.info(configStr.toString());
+    }
+  }
+
+  /**
+   * Can't create automatically during inserting, so create one if necessary.
+   * @param table
+   */
+  private void createCollection(String table) {
+    synchronized(ArangoDBClient.class) {
+      if (defaultCollection == null) {
```

Comment:
this isn't going to properly handle when the table name changes across calls. the defaultCollection should be something like the last table accessed and it should be per DB instance.